enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,437 @@
+all the router related resources, including APIs and controllers.
+
+Setting the option to `Enabled` will deploy the router, while `Disabled` will disable
+it. Default will be `Enabled`.
+
+### Listening ports
+The following configuration is proposed:
+```yaml
+ingress:
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+```
+
+In order to allow this configuration and many other advantages, the
+router service shall be exposed using `LoadBalancer` type. See the
+Design Details section for more information.
+
+#### Firewalling ports
+Using `LoadBalancer` service type prevents the usage of firewalld to block
+access to ports. Since this is a feature that users may require for auditing,
+an alternative is provided.
+
+The `NetworkPolicy` resource fits the needs for blocking traffic to services
+and pods. Policies can also be configured to allow specific IP addresses and
+provide a high degree of customization for network traffic. They use the CNI
+instead of host level settings, providing greater flexibility and being self
+contained within MicroShift.
+
+Users in need of blocking traffic shall use the `NetworkPolicy` resource. A
+section in the documentation will be added to describe how to do this.
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+  expose:
+    - <IP address/NIC> # Defaults to the NICs of the node
+```
+
+As we have seen in previous sections, the use of `LoadBalancer` makes ovnk
+configure iptables rules to expose the service outside of the cluster.
+Building on [LoadBalancer controller enhancement](loadbalancer-service-support.md),
+the service is exposed using the node IP and configured ports.
+
+All the IPs where the router is exposed need to be part of the `LoadBalancer`
+service status definition. In order to provide more flexibility, the
+configuration options are able to take IP addresses and/or NIC names in the
+system. Using NIC names abstracts away IP changes, in cases where this
+happens.
+
+As a default value, MicroShift will take the NIC names for each of the default
+routes found in the system.
+
+### Allow specific IP addresses
+This may be seen as a special case of firewalling, but using IP addresses
+instead of ports. Using the same `NetworkPolicy` resources, IPs may be
+specified to allow traffic from them and reject it from any other source.
+
+Please note this is a manual procedure for the user. Documentation will be
+added on how to perform the required actions.
+
+For example, the following policy allows traffic only from CIDR
+`192.168.122.0/24` to port 443 (even internal traffic would be rejected):
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  creationTimestamp: "2023-10-31T13:17:42Z"
+  generation: 2
+  name: block-router
+  namespace: openshift-ingress
+  resourceVersion: "5350"
+  uid: e00f5ec7-1557-4285-aa1c-b1b909e53b7a
```

Comment:
Just a nit, but you could drop the cluster-managed fields since they aren't meant to be user facing.  That would just leave `name` and `namespace`, i.e. the user facing fields.